### PR TITLE
improved get_application and minor changes to city_state

### PR DESCRIPTION
**Pull request by torarm**
I added method to BIACase to return the U.S city and state of the original application.

- Uses regex to find city, state abbreviation 5-digit zip code pattern. Not sure if it's the most effective way to find it, I found the lawyer's mailing address is usually the same as the applicant's location.
- Lawyer mailing address format is the same for application decisions and appeals. The method should safeguard against Falls Church, Virginia
- Edge case: Lawyer being located outside the state of application

### CODE REVIEW

File: app/ocr.py [hunk 1]
```diff
@@ -299,6 +301,18 @@ def get_outcome(self) -> str:
...
+        locs = re.findall("(?:[a-zA-Z ]{1,20}, [A-Z]{2} \d{5})", self.doc.text)
+        gpe = set(str(ent) for ent in self.doc.ents if ent.label_ == 'GPE')
+        for i in gpe:
+            for x in locs:
+            if i in x:
+                s = " ".join(locs)
+                return str(re.findall("(?:%s, [A-Z]{2})"%i, s)[0])
 
     def get_based_violence(self) -> Union[Dict[str, List[str]], None]:
         """
```
Comment (JenFaith) [inline app/ocr.py:347]: This looks good to me, but should we split up state/city?

One other thing just as an idea, @torarm, could you use the get_ents function that's already defined to simplify some of the code a little?

**Review by Daniel-Fernandez-951 — APPROVED** [review]
Looks great.

**Comment by BrokenShell** [comment]
This finds the first state abbreviation that it finds. Is this ALWAYS the right state? What about city?

**Comment by torarm** [comment]
This pull request has been updated.
- I scraped the locations of all the immigration courts (there's 69 courts, file has 61, some cities have more than one court) and converted to JSON
- The file reads in the JSON file as a dictionary
- Checks the Spacy GPE label to match a city in the dictionary
- Returns the state and city in separate fields.

**Comment by brett-doffing** [comment]
> Can someone remind me what `gpe` is?

If I'm not mistaken, it means Geo Political Entity @KWeatherwalks

**Review by ren-curry — APPROVED** [review]
This code looks good. I like the changes to get_application restructuring the if statements to be simpler. Additionally the change to the state and city, I think the change to split them was an important addition.

**Review by Daniel-Fernandez-951 — APPROVED** [review]
Great work and thank you for splitting up city and state. We will need to update the BE team and our [documentation to reflect these changes](https://trello.com/c/GSquBOUp).

**Pull request merged**